### PR TITLE
feat: add buf.build example

### PR DESCRIPTION
**Pull request by thesayyn**
### Type of change

- New feature or functionality (change which adds functionality)

### Test plan

- New test cases added

### CODE REVIEW

**Comment by alexeagle** [comment]
I was expecting one that uses the buf npm package like that client who asked us about it, rather than an added toolchain. Is that the wrong way to do it, or just a different example?

**Comment by thesayyn** [comment]
> I was expecting one that uses the buf npm package like that client who asked us about it, rather than an added toolchain. Is that the wrong way to do it, or just a different example?

that'd be rules_js + buf.build example which is far different than this one here.

> Is that the wrong way to do it,

I can't say "wrong" but suboptimal as you have to workaround many things like plugin paths and output paths.

**Comment by alexeagle** [comment]
okay no need to block this one, but the JS one with https://github.com/bufbuild/protobuf-es is the example I'm really interested in.

At that client I think I resolved the problem with plugin paths, at least partway.

File: bufbuild/WORKSPACE
Comment (alexeagle) [inline bufbuild/WORKSPACE:1]: Can we make new examples bzlmod from the start with an empty WORKSPACE.bazel?

File: bufbuild/buf.bzl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    args = ctx.actions.args()
+    args.add("generate")
+    args.add(ctx.file.template, format = "--template=%s")
+
+    inputs = [ctx.file.template]
+    outputs = []
+    for dep in ctx.attr.deps:
+        outputs.extend(
+            proto_common.declare_generated_files(
```
Comment (seungduk-yanolja) [inline bufbuild/buf.bzl:14]: Should it be Go-specific? Would it be possible you guide us how to generate pb files for other languages?

File: bufbuild/buf.bzl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ctx.actions.run(
+        executable = buf_toolchain.buf_info.binary,
+        inputs = inputs,
+        arguments = [args],
+        outputs = outputs,
+    )
+
+    return DefaultInfo(files = depset(outputs))
+
+buf_proto_library = rule(
```
Comment (seungduk-yanolja) [inline bufbuild/buf.bzl:34]: Maybe buf_go_proto_library or something not to confuse readers

File: bufbuild/buf.bzl (hunk 1 above)
Comment (alexeagle) [reply bufbuild/buf.bzl:14]: Yeah this example is Go-specific. We can definitely show you other languages, but we don't have time to do that for free. Consider https://www.aspect.dev/bazel-open-source-support or fund with the Sponsor link on this repo.

File: bufbuild/buf.bzl (hunk 2 above)
Comment (alexeagle) [reply bufbuild/buf.bzl:34]: agreed

**Review by alexeagle — APPROVED** [review]

**Pull request merged**